### PR TITLE
Deprecate `SplitButton` and remove usages

### PR DESCRIPTION
**Pull request by thomasdax98**
- Deprecate `SplitButton`, `FinalFormSaveSplitButton` and `SplitButtonContext`
- Removes all usages of these components in the libraries and Demo
- Adds a warning to the Docs and removes some of the stories concerning `SplitButton`

---

Closes COM-375 and COM-393

### CODE REVIEW

File: packages/admin/admin-stories/src/docs/components/SplitButton/SplitButton.stories.mdx [hunk 1]
```diff
@@ -1,71 +0,0 @@
-import { Meta, Story, Canvas } from "@storybook/addon-docs";
```
Comment (thomasdax98) [inline packages/admin/admin-stories/src/docs/components/SplitButton/SplitButton.stories.mdx:1]: I removed the Docs on the `SplitButton` together with all stories. But I'm not sure if that's a good idea. Maybe we should keep at least one story for development if we, for some reason, need to change something about the `SplitButton`

**Review by johnnyomair — COMMENTED** [review]
I don't think we should remove the docs, since we're "only" deprecating the feature, but not removing it. The docs should have a deprecation notice however.

**Comment by thomasdax98** [comment]
@johnnyomair I readded the stories and docs and added a deprecation notice

**Comment by johnnyomair** [comment]
@thomasdax98 Storybook was moved to the root, please update this PR. Thanks!

File: packages/admin/admin-stories/src/docs/components/SplitButton/SplitButton.stories.mdx (hunk 1 above)
Comment (johnnyomair) [reply packages/admin/admin-stories/src/docs/components/SplitButton/SplitButton.stories.mdx]: Resolved with https://github.com/vivid-planet/comet/pull/1623#issuecomment-1926405750

**Pull request merged**